### DISCUSSION TITLE
C# V2.0 SDK - LWA Credentials

### DISCUSSION THREAD

**Discussion by kevnhowe** [amzn/selling-partner-api-models#3721] [General]
I am migrating from v1 of the c# sdk with aws authentication and have built the C# v2 SDK following the instructions on the "Automate your sp-api calls..." article.
I cannot seem to apply lwa credentials correctly to the api call. The instructions make it seem very straight forward. However, the instructions also reference a Builder method that is not a part of any of the APIs I have built, including the Sellers Api in the example.
Can you please elaborate on the method used to connect via the new SDK using the sime lwa authentication credentials?

**Comment by kevnhowe** [comment]
I figured this out, but think it would be good to share with others trying to get this working. I had to call the APIAA library directly to request the access token for the given lwa credentials. once I had the Access Token, I had to add a header value for the access token to the API request configuration.
It seems as if the Client API does not fully support the APIAA yet, as it does not call the lwa api directly and apply the access token to the request.

**Reply by kevnhowe** [reply]
This has been solved partially. I have been able to connect and transact with the API using the new sdk with my test console app, but cannot transact via my test excel vsto add-in. The addin uses the same sdk. It can successfully acquire an access token via the apiaa API, but I get a 400 error whenever I request any API object. I have troubleshot and confirmed that the request looks exactly like the one sent by the successful console app.
The excel addin was previously working in production using the old version of the sdk with aws credentials. I am trying to update to use the new sdk.

**Comment by MeenaAmz** [comment]
@kevnhowe Can you please elaborate which api are you attempting to call through sdk? Once you generate c# sdk using swagger code gen, every api would have its own builder method and all you have to do is reference the AA library and it should work. Please make sure you have followed all the steps mentioned [here](https://developer-docs.amazon.com/sp-api-blog/docs/automate-your-sp-api-calls-using-the-c-sdk).

**Reply by kevnhowe** [reply]
I have followed the instructions you linked. I have tried several times. I have the Sellers API included in my project. However, there is no Builder method in the class at all.
![image](https://github.com/amzn/selling-partner-api-docs/assets/18247661/2dcc0f9a-e2d5-4257-a597-262634314b97)
None of the APIs generated from the swagger code have a builder method.

**Reply by MeenaAmz** [reply]
Builder is not a method within SellersApi class, its actually a class nested inside the SellersApi class. It would look something like this. 
![image](https://github.com/amzn/selling-partner-api-docs/assets/115670748/c4905038-8363-4367-8175-4da7ae59f80e)
To access it, you will have to use this code. The code is similar for other APIs as well.
 SellersApi sellersApi = new SellersApi.Builder()
                    .SetLWAAuthorizationCredentials(lwaAuthorizationCredentials)
                    .Build();

**Reply by kevnhowe** [reply]
Thanks for the reply. however 
![image](https://github.com/amzn/selling-partner-api-docs/assets/18247661/d20ed3dc-511b-4df8-95e4-fb3f5a77bb00)
which version of the API are you referring to? That Builder class does not exist in the API I have generated from the latest version.
